lib/Doctrine/DBAL/Types/DateIntervalType.php

```diff
@@ -57,9 +57,16 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
         }
 
         try {
-            $interval = new \DateInterval(substr($value, 1));
+            $negative = false;
 
-            if (substr($value, 0, 1) === '-') {
+            if ($value[0] === '+' || $value[0] === '-') {
```

Comment:
The value may be an empty string, so before it gets converted to an interval, we'll have a notice. Probably, the previous approach with `substr()` should be used instead.

Reply:
Used isset.
Note that using `substr('', 1)` produces FALSE, not empty string, but no warning. #php